pkg/kube-metrics/store.go

```diff
@@ -30,15 +30,23 @@ import (
 // NewMetricsStores returns collections of metrics in the namespaces provided, per the api/kind resource.
 // The metrics are registered in the custom metrics.FamilyGenerator that needs to be defined.
 func NewMetricsStores(dclient dynamic.NamespaceableResourceInterface, namespaces []string, api string, kind string,
```

Comment:
@raffaelespazzoli It's a little weird passing `namespaces` to this function when `namespaced` is false.
WDYT about having separate `NewMetricsStores` functions? One for namespaced resources and another for cluster-scoped resources? For example:

```
// Deprecated: ...
func NewMetricsStores() {
    return NewNamespacedMetricsStores()
}


func NewNamespacedMetricsStores() {
	// This function is what was NewMetricsStores, just renamed.
}

func NewClusterScopedMetricsStores() { ... }
```

Reply:
done.